### ISSUE TITLE
Lack of Information on tools for creating/modifying container images

### ISSUE THREAD

**Issue by harridu** [kubernetes/website#41957]
This page lacks an important piece of information: What is the recommended way to either create a new container image or to modify an existing one for using it in Kubernetes. Downloading an image somebody else created might not be sufficient.

A pointer to external projects (e.g. [kaniko](https://github.com/GoogleContainerTools/kaniko) or [buildah](https://github.com/containers/buildah)) might be sufficient.

**Comment by dipesh-rawat** [comment]
_Page related to issue_: https://kubernetes.io/docs/concepts/containers/images/

**Comment by dipesh-rawat** [comment]
/retitle: Lack of Information on tools for creating/modifying container images

**Comment by dipesh-rawat** [comment]
Mistyped ":" earlier!

/retitle Lack of Information on tools for creating/modifying container images

**Comment by dipesh-rawat** [comment]
When it comes to creating or modifying container images for use in Kubernetes, there are indeed multiple approaches we can take. While downloading pre-existing images can be a starting point, it may not always meet our specific requirements.
To address this issue, we can add links to relevant image building tools as suggested _(or pointers to any existing page(s) that provide information on image building)_. 
I will let other members/contributors share their thoughts on this.

**Comment by sftim** [comment]
Ideally, we find a CNCF page on this topic and link to that page (so we do not need to maintain our own).

I'm imagining something like https://collabnix.com/docker-tools-a-curated-list-of-docker-developer-tools/ but backed by the CNCF.
If that CNCF pages doesn't exist, maybe we can help change that&mdash;and then link there.

**Comment by shannonxtreme** [comment]
I agree with @sftim and also thoughts: documenting creating and modifying images is out of scope for the Kubernetes project IMO, but would be cool to link to

**Comment by sftim** [comment]
CNCF's docs team may be willing to lead on creating something we can link to.

**Comment by mrgiles** [comment]
This CNCF Landscape section: [Application Definition & Image Build](https://landscape.cncf.io/?category=application-definition-image-build&grouping=category) lists several image-building tools (Kaniko, KO, Podman, etc.) but unfortunately, they're grouped among several other tools, so it's not a very specific resource.

**Comment by mrgiles** [comment]
What do you think @nate-double-u, could the CNCF Landscape satisfy this request?

**Comment by nate-double-u** [comment]
Hi @mrgiles, I'm afraid I haven't had a chance to look too closely at this yet.

I'll be away on vacation next week, but will take a look when I'm back.

**Comment by nate-double-u** [comment]
Sorry it's taken me a while to circle back to this. I don't think this is something that the CNCF can write or host on it's site. If the CNCF starts pointing to external projects (and non CNCF projects) then we start getting into non-vendor-neutral territory.

I think the answer here is to find examples of what is needed in existing CNCF CRI projects, and if that doesn't exist, then the community here should reach out and talk to those projects to get things listed.

**Comment by sftim** [comment]
This isn't about CRI (a Kubernetes interface to daemons like containerd and CRI-O). Those are tools for _running_ containers.

This issue is about explaining how to _build_ containers. I want to emphasize that the CNCF does include software projects for container image build. That could be `nerdctl build` (using containerd, a CNCF project) or `pack` from buildpacks.io (also CNCF). We could mention that, in the ecosystem, there might be other players - I know that Docker and Podman are both popular. Then there's a choice to provide a basic list of wider ecosystem tools, or to only mention CNCF projects.

**Comment by mrgiles** [comment]
Going back to my original post, I think the CNCF already provides some links to image-building software in their [Cloud Native Interactive Landscape](https://landscape.cncf.io/card-mode?category=application-definition-image-build&organization=cloud-native-computing-foundation-cncf&grouping=category) (this URL was generated using some filters provided in the search panel). I'm sure it's not comprehensive but it provides a starting point. 

Using it, I was able to find these 4 related projects: [Buildpack.io](https://buildpacks.io/), [kbld](https://carvel.dev/kbld/), [DevSpace](https://www.devspace.sh/), [Ko](https://ko.build/). These are listed as CNCF projects. There are other projects from private companies listed as well. This is always going to be a moving target since new Kubernetes-related projects appear all the time but, like I said, it provides a starting point and it's already there, no need to build (and maintain) a new list.

@nate-double-u, @sftim

**Comment by tengqm** [comment]
The removal of Docker as container runtime did confused quite some users.
We may need to provider some pointers to CNCF projects so that users know where to look at,
as @sftim mentioned above.
This doesn't mean we will dual-host those detailed docs about the tools.

**Comment by sftim** [comment]
(if we document `nerdctl build`, we can eg let people know that `podman build` and `docker build` work similarly)
